File: frontend/src/types/armor.ts
export interface SkillLevel {
  level:       number
  description: string | null
}

export interface ArmorSkill {
  name:        string
  description: string | null
  level:       number | null
  levels:      SkillLevel[]
}

export interface ArmorMaterial {
  itemId:   number
  name:     string
  quantity: number | null
}

export interface ArmorSetBonusSkill {
  name:        string
  description: string | null
  required:    number | null
  levels:      SkillLevel[]
}

export interface ArmorSetBonus {
  name:        string
  description: string | null
  skills:      ArmorSetBonusSkill[]
}

export interface ArmorPiece {
  id:     number
  name:   string
  type:   'head' | 'chest' | 'arms' | 'waist' | 'legs' | string
  rarity: number | null

  defenseBase:       number | null
  defenseMax:        number | null
  defenseAugmentMax: number | null

  fire:    number | null
  water:   number | null
  thunder: number | null
  ice:     number | null
  dragon:  number | null

  slot1: number | null
  slot2: number | null
  slot3: number | null

  skills:    ArmorSkill[]
  materials: ArmorMaterial[]
}

export interface ArmorSet {
  id:     number
  name:   string
  rank:   'LR' | 'HR' | 'MR' | string
  bonus:  ArmorSetBonus | null
  pieces: ArmorPiece[]
}

File: backend/src/main/java/com/asteradatabank/armor/Armor.java
package com.asteradatabank.armor;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "armor")
@Getter
@NoArgsConstructor
public class Armor {

    @Id
    private Integer id;

    @Column(name = "order_id")
    private Integer orderId;

    private Integer rarity;

    private String rank;

    // "head" | "chest" | "arms" | "waist" | "legs"
    @Column(name = "armor_type")
    private String armorType;

    @Column(name = "armorset_id")
    private Integer armorsetId;

    @Column(name = "recipe_id")
    private Integer recipeId;

    @Column(name = "slot_1") private Integer slot1;
    @Column(name = "slot_2") private Integer slot2;
    @Column(name = "slot_3") private Integer slot3;

    @Column(name = "defense_base")        private Integer defenseBase;
    @Column(name = "defense_max")         private Integer defenseMax;
    @Column(name = "defense_augment_max") private Integer defenseAugmentMax;

    private Integer fire;
    private Integer water;
    private Integer thunder;
    private Integer ice;
    private Integer dragon;
}
